### PR TITLE
Fixed acc to mentor review

### PR DESCRIPTION
**Pull request by Zel1oy**

### CODE REVIEW

File: src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    private static ReadFileService fileReader;
+
+    @BeforeAll
+    static void setUp() {
+        fileReader = new ReadFileFromCsv();
+    }
+
+    @Test
+    public void read_ValidFile_Ok() {
+        List<String> lines = fileReader.readFromFile("src/test/resources/test_input.csv");
```
Comment (ihorvasylenko23) [inline src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java:22]: Maybe you should use costant for filePath

File: src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+        Assertions.assertNotNull(lines);
+        assertEquals(3, lines.size());
+    }
+
+    @Test
+    public void read_NonexistentFile_ExceptionThrown() {
+        RuntimeException runtimeException = assertThrows(
+                RuntimeException.class, () -> fileReader.readFromFile("invalid_file.csv"));
```
Comment (ihorvasylenko23) [inline src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java:30]: Maybe you should use costant for invalid filePath

File: src/test/java/core/basesyntax/impl/ReportCreatorTest.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportCreator;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReportCreatorTest {
```
Comment (ihorvasylenko23) [inline src/test/java/core/basesyntax/impl/ReportCreatorTest.java:11]: Maybe u need more tests, for example with empty data

File: src/test/java/core/basesyntax/impl/WriteToFileDataToCsvTest.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.service.WriteDataToFileService;
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class WriteToFileDataToCsvTest {
```
Comment (ihorvasylenko23) [inline src/test/java/core/basesyntax/impl/WriteToFileDataToCsvTest.java:12]: Maybe you should add test to check thrown exception

File: src/test/java/core/basesyntax/impl/ReportCreatorTest.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    static void setUp() {
+        reportCreator = new ReportCreatorImpl();
+    }
+
+    @Test
+    void createReport_ValidData_Ok() {
+        Storage.getStorage().put("apple", 25);
+        Storage.getStorage().put("banana", 15);
+        String report = reportCreator.prepare();
+        String expectedReport = "fruit,quantity\nbanana,15\napple,25";
```
Comment (ihorvasylenko23) [inline src/test/java/core/basesyntax/impl/ReportCreatorTest.java:24]: This can be taken out in a constant

File: src/test/java/core/basesyntax/model/FruitTransactionTest.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package core.basesyntax.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+class FruitTransactionTest {
```
Comment (ihorvasylenko23) [inline src/test/java/core/basesyntax/model/FruitTransactionTest.java:7]: according to the task, only services should be tested, but I think you can leave the tests for FruitTransaction class if the mentors are not opposed

File: src/test/java/core/basesyntax/model/OperationTest.java [hunk 7]
```diff
@@ -0,0 +1,27 @@
+package core.basesyntax.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.Test;
+
+class OperationTest {
```
Comment (ihorvasylenko23) [inline src/test/java/core/basesyntax/model/OperationTest.java:8]: according to the task, only services should be tested, but I think you can leave the tests for Operation class if the mentors are not opposed

**Review by ihorvasylenko23 — CHANGES_REQUESTED** [review]
Good job) Added some suggestions

**Review by ihorvasylenko23 — APPROVED** [review]

File: src/test/java/core/basesyntax/impl/DataConverterToObjectTest.java [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+        inputData.add("type,fruit,quantity");
+        inputData.add("b,apple,10");
+        List<FruitTransaction> expected = new ArrayList<>();
+        expected.add(new FruitTransaction(Operation.BALANCE, "apple", 10));
+        List<FruitTransaction> actual = dataConverter.convert(inputData);
+        assertTrue(expected.equals(actual));
+    }
+
+    @Test
+    void convert_NullData_NotOk() {
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/impl/DataConverterToObjectTest.java:34]: please make from this test two tests, bacause it is not unit test now

File: src/test/java/core/basesyntax/impl/OperationProcessTest.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+import core.basesyntax.strategy.ReturnOperationHandler;
+import core.basesyntax.strategy.SupplyOperationHandler;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class OperationProcessTest {
+    private static DataProcesser dataProcesser;
+    private static List<FruitTransaction> fruitTransactionList;
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/impl/OperationProcessTest.java:22]: You don`t need this variable on class level

File: src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java [hunk 10]
```diff
@@ -0,0 +1,36 @@
...
+
+    @BeforeAll
+    static void setUp() {
+        fileReader = new ReadFileFromCsv();
+    }
+
+    @Test
+    public void read_ValidFile_Ok() {
+        List<String> lines = fileReader.readFromFile(VALID_FILE_PATH);
+        Assertions.assertNotNull(lines);
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java:25]: please use static import

File: src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+        assertEquals(3, lines.size());
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/impl/ReadFileFromCsvTest.java:26]: Is it enough to test that size is as expected maybe it makes sence to compare content as well

File: src/test/java/core/basesyntax/impl/ReportCreatorTest.java [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.service.ReportCreator;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+class ReportCreatorTest {
+    private static final String EXPECTED_REPORT = "fruit,quantity\nbanana,15\napple,25";
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/impl/ReportCreatorTest.java:12]: please make this local variable or add  String expectedReport = "fruit,quantity"; on class level too

File: src/test/java/core/basesyntax/impl/WriteToFileDataToCsvTest.java [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+            String line = bufferedReader.readLine();
+            assertEquals(report, line, "Content should match what was written!");
+        } catch (IOException e) {
+            throw new RuntimeException("Can't read from file: " + VALID_FILE_NAME, e);
+        }
+    }
+
+    @Test
+    public void testWriteToFile_Exception() {
+        assertThrows(RuntimeException.class,
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/impl/WriteToFileDataToCsvTest.java:39]: Please make sure that error message correct as well

File: src/test/java/core/basesyntax/impl/WriteToFileDataToCsvTest.java [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+    @BeforeAll
+    static void setUp() {
+        writeDataToFileService = new WriteToCsv();
+    }
+
+    @Test
+    void write_ValidData_Ok() {
+        String report = "Test report data";
+        writeDataToFileService.writeToFile(report, VALID_FILE_NAME);
+        try (BufferedReader bufferedReader = new BufferedReader(new FileReader(VALID_FILE_NAME))) {
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/impl/WriteToFileDataToCsvTest.java:29]: Please move reading logic to separate method

File: src/test/java/core/basesyntax/model/OperationTest.java [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+
+    @Test
+    void convert_UnexpectedCode_To_Operation_NotOk() {
+        assertThrows(RuntimeException.class,
+                () -> Operation.convertToOperation("invalid"));
+    }
+
+    @Test
+    void convert_ValidCode_Ok() {
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/model/OperationTest.java:17]: It is not unit test

File: src/test/java/core/basesyntax/strategy/BalanceOperationHandlerTest.java [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+    @Test
+    void handle_ValidData_Ok() {
+        balanceHandler.handle("banana", 20);
+        assertTrue(Storage.getStorage().containsKey("banana"));
+        assertEquals(20, (int) Storage.getStorage().get("banana"));
+    }
+
+    @Test
+    void handle_NegativeData_NotOk() {
+        assertThrows(RuntimeException.class,
```
Comment (andrii-hoienko) [inline src/test/java/core/basesyntax/strategy/BalanceOperationHandlerTest.java:34]: make sure error message is correct as well

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
Nice, but some improvements required

File: src/test/java/core/basesyntax/model/FruitTransactionTest.java (hunk 6 above)
Comment (vovasalyha) [inline src/test/java/core/basesyntax/model/FruitTransactionTest.java:7]: You don't need to test plain model classes.

**Review by vovasalyha — CHANGES_REQUESTED** [review]
Please remove .DS_Store files from the project, if they appear anyway maybe consider adding it to .gitignore

**Review by vovasalyha — APPROVED** [review]
